### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by a-laarabi**
> # In this pull request I created:

- create module users
- create module posts
- create module likes
- create module comments

### CODE REVIEW

File: app/models/comments.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Comments < ApplicationRecord
+  belongs_to :users, foreign_key: 'author_id', class_name: 'User'
```
Comment (DeliceLydia) [inline app/models/comments.rb]: - [ ] Please remember that `belongs_to` should use a singular term. If you use the pluralized form in the above example for the `user` association in the Comment model and try to create the instance by `Comment. create(users: @user),` you would be told that there was an "uninitialized constant Comment::Users".
so it should be `belongs_to :user`

File: app/models/comments.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  belongs_to :posts, foreign_key: 'post_id', class_name: 'Post'
```
Comment (DeliceLydia) [inline app/models/comments.rb]: - [ ] Same here it should be `belongs_to :post`

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class Like < ApplicationRecord
+  belongs_to :users, foreign_key: 'author_id', class_name: 'User'
```
Comment (DeliceLydia) [inline app/models/like.rb]: - [ ] It should be `belongs_to :user`

File: app/models/like.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  belongs_to :posts, foreign_key: 'post_id', class_name: 'Post'
```
Comment (DeliceLydia) [inline app/models/like.rb]: - [ ] `belongs_to :post`

File: app/models/like.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  after_save :update_likes_counter
+
+  private
+
+  def update_likes_counter
+    posts.increment!(:likes_counter)
+  end
```
Comment (DeliceLydia) [inline app/models/like.rb]: - [ ] This should be `post.increment!`

File: app/models/comments.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+  after_save :update_comments_counter
+
+  validates :text, presence: true
+
+  def update_comments_counter
+    posts.update(comments_counter: posts.comments_counter + 1)
```
Comment (DeliceLydia) [inline app/models/comments.rb]: - [ ] Same here you can make it `post.update` etc

File: app/models/post.rb [hunk 7]
```diff
@@ -0,0 +1,18 @@
+class Post < ApplicationRecord
+  has_many :comments
+  has_many :likes
+  belongs_to :users, foreign_key: 'author_id', class_name: 'User'
```
Comment (DeliceLydia) [inline app/models/post.rb]: - [ ] This should be `belongs_to :user` as well you can have a look at this [link](https://guides.rubyonrails.org/association_basics.html)

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @a-laarabi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! Highlights:
- Followed git workflow
- No linter errors


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @a-laarabi 

Your project is complete! Great job for making the requested changes🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- Validations correctly implemented
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by a-laarabi** [comment]
Thank you @DeliceLydia and @tufoinnkuo10

**Pull request merged**